ann_benchmarks/datasets.py

```diff
@@ -34,18 +33,24 @@ def get_dataset(which):
             print("Creating dataset locally")
             DATASETS[which](hdf5_fn)
     hdf5_f = h5py.File(hdf5_fn, 'r')
-    return hdf5_f
+
+    dimension = hdf5_f.attrs['dimension'] if 'dimension' in hdf5_f.attrs else len(hdf5_f['train'][0])
```

Comment:
I would like to see a comment here that this is for reasons of backwards compatibility.